relayer/types/types.go

```diff
@@ -165,6 +165,14 @@ func (m *MessageCache) Remove(key *MessageKey) {
 	delete(m.Messages, *key)
 }
 
+// Get returns the message from the cache
+func (m *MessageCache) Get(key *MessageKey) (*RouteMessage, bool) {
+	m.Lock()
```

Comment:
Would only RLock make sense here instead of Lock ? @debendraoli

Reply:
yup, worth a try